### DISCUSSION TITLE
how to get started for gsoc

### DISCUSSION THREAD

**Discussion by gintama91** [scarpe-team/scarpe#138] [Q&A]
hey, @noahgibbs 
is there any way to contact mentors for this project as per gsoc 2023?
i would like to know if there are any good first issues that to be done or how to get around with codebase 

Thankyou

**Comment by noahgibbs** [comment]
Hello! This is a great start. We have a project Discord, which would also be an important resource. Let me talk to @schwad about how to get you an invitation to that. You can see at least one "good first issue" in the project issues, but it's probably a great idea to talk to somebody first and get a quick tour of the codebase.

**Reply by noahgibbs** [reply]
Here we go - issues marked as "good first issue": https://github.com/scarpe-team/scarpe/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

**Reply by noahgibbs** [reply]
Hm. If we can message you somehow, we can send out a Discord server invitation. I don't know if we're putting that up publicly, though.

**Reply by gintama91** [reply]
hey thanks man! yeah i will get arnd in discord and will understand codebase first instead of jumping right into issues.

**Comment by Schwad** [comment]
Hi @gintama91 ! This link should get you into our discord, come say hi! https://discord.gg/nbhaZJtfVW

It's great to start by forking the project and trying to run examples in the examples directory. e.g. 

`./exe/scarpe examples/hello_world.rb --dev`

And see them work!

Quick things you could do is try this command out with examples in the "not_working" directory and see if they're working. Also, you could level up some of your knowledge by building a few desktop apps yourself and adding them as examples in the repo! (We're always looking for more working examples :) )

As you get comfortable with that, you might want to implement something. For new features, we are following the NOBODY KNOWS SHOES manual and also a larger more formal manual to follow the original DSL/features. Basically, we want everything in [this manual](https://github.com/scarpe-team/scarpe/blob/main/docs/static/manual.md) to work.

The README could use a bit of updating and clarifying on how we got on on the hackdays, I'll work on that soon.

Currently, everything talks to "webview" (with HTML/JS/CSS) to build desktop apps. That's our most built-out backend (we call it a display service) and is the default.

Another thing we're exploring, if you get interested down the line, is alternative display services. Right now I'm playing with [Glimmer](https://github.com/AndyObtiva/glimmer-dsl-libui), so basically ruby-to-ruby instead of ruby-to-html.

I think you'll find it very fun and exciting to have working desktop apps right away and build on top of that. Let me know if you have any questions!

**Reply by gintama91** [reply]
oh thanks i will look into that!

**Reply by sapienfrom2000s** [reply]
@Schwad discord invitation link has expired

**Reply by noahgibbs** [reply]
Here's a newer one:

https://discord.gg/6UAhvnVUJz
